src/OpenApi/src/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class OpenApiSchemaExtensions
+{
+    /// <summary>
+    /// Generates a deep copy of a given <see cref="OpenApiSchema"/> instance.
+    /// </summary>
+    /// <remarks>
+    /// The copy constructors of the <see cref="OpenApiSchema"/> class do not perform a deep
+    /// copy of the instance which presents a problem whe making modifications in deeply nested
+    /// subschemas. This extension implements a deep copy on <see cref="OpenApiSchema" /> to guarantee
+    /// that modifications on cloned subschemas do not affect the original subschema.
+    /// /// </remarks>
+    /// <param name="schema">The <see cref="OpenApiSchema"/> to generate a deep copy of.</param>
+    public static OpenApiSchema Clone(this OpenApiSchema schema)
+    {
+        return new OpenApiSchema
+        {
+            Title = schema.Title,
+            Type = schema.Type,
+            Format = schema.Format,
+            Description = schema.Description,
+            Maximum = schema.Maximum,
+            ExclusiveMaximum = schema.ExclusiveMaximum,
+            Minimum = schema.Minimum,
+            ExclusiveMinimum = schema.ExclusiveMinimum,
+            MaxLength = schema.MaxLength,
+            MinLength = schema.MinLength,
+            Pattern = schema.Pattern,
+            MultipleOf = schema.MultipleOf,
+            Default = OpenApiAnyCloneHelper.CloneFromCopyConstructor<IOpenApiAny>(schema.Default),
+            ReadOnly = schema.ReadOnly,
+            WriteOnly = schema.WriteOnly,
+            AllOf = schema.AllOf?.Select(s => s.Clone()).ToList(),
+            OneOf = schema.OneOf?.Select(s => s.Clone()).ToList(),
+            AnyOf = schema.AnyOf?.Select(s => s.Clone()).ToList(),
+            Not = schema.Not?.Clone(),
+            Required = schema.Required != null ? new HashSet<string>(schema.Required) : null,
+            Items = schema.Items?.Clone(),
+            MaxItems = schema.MaxItems,
+            MinItems = schema.MinItems,
+            UniqueItems = schema.UniqueItems,
+            Properties = schema.Properties?.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.Clone()),
+            MaxProperties = schema.MaxProperties,
+            MinProperties = schema.MinProperties,
+            AdditionalPropertiesAllowed = schema.AdditionalPropertiesAllowed,
+            AdditionalProperties = schema.AdditionalProperties?.Clone(),
+            Discriminator = schema.Discriminator != null ? new(schema.Discriminator) : null,
+            Example = OpenApiAnyCloneHelper.CloneFromCopyConstructor<IOpenApiAny>(schema.Example),
+            Enum = schema.Enum != null ? new List<IOpenApiAny>(schema.Enum) : null,
```

Comment:
Is this different from `schema.Enum?.ToList()`?

Reply:
It seems that they are equivalent based on the fact that OpenApiSchema.Enum is `IList<IOpenApiAny>` and the [current implementation of `ToList`](https://source.dot.net/#System.Linq/System/Linq/ToCollection.cs,60). I find `new()` to be a bit clearer about what is happening here.

Reply:
Fine by me, but `AllOf` etc use `ToList`.